### PR TITLE
Add HDR calibration to GI settings page

### PR DESCRIPTION
**Pull request by shatyuka**
Adds HDR calibration images to the settings page:
screen shot

Other adjustments:
- Hide HDR calibration panel by default, for it's much bigger than before.
- Little changes of help tooltips.

### CODE REVIEW

**Comment by bagusnl** [comment]
Can I get a confirmation that what shown in the Collapse calibration page is similar (or similar enough) as the one in-game?
I don't have access to HDR monitor so I couldn't test sorry

**Comment by MapleHinata** [comment]
> Can I get a confirmation that what shown in the Collapse calibration page is similar (or similar enough) as the one in-game? I don't have access to HDR monitor so I couldn't test sorry

I tested it on my display and it looks very close (if not identical) to what the game shows. 
Nice job, @shatyuka.

**Comment by bagusnl** [comment]
Thank you for the confirmation @MapleHinata !
My only bit of gripe is that the 1st HDR Calibration image is square so it looks a bit off in my eyes. Can it have a white background to match the width of the other images as well?

For the other code, I've done a short look through and all seems good, will do a thorough look into it soon

**Comment by shatyuka** [comment]
@bagusnl Is this ok?

**Comment by bagusnl** [comment]
Looks fine on me IMO

**Comment by bagusnl** [comment]
I checked out your branch and tried to build it. It seems that the 1st panel image is broken somehow and keep only displaying white on me. I tested both on Debug and Publish and same behavior seen. It does spat out some ArgumentException and IOException in the debug output.
```
System.ArgumentException: Value does not fall within the expected range.
   at CollapseLauncher.Pages.GenshinGameSettingsPage.GetAppFileAsync(Uri uri) in F:\GitHub\Collapse\CollapseLauncher\XAMLs\MainApp\Pages\GenshinGameSettingsPage.xaml.cs:line 394
   ```
image

**Comment by shatyuka** [comment]
Actually the first image is just a white icon displayed on a white background, and the slider controls the brightness of the icon. So it appears completely white on SDR monitor.
For the ArgumentException, I have a try-catch block to handle it. It may be because your debug mode takes over exception handling.
`ms-appx://` uri failed to be opened in c# code, but works in xaml for an unpackaged app. Maybe there is a better way to detect whether this application is packaged? IDK

**Comment by bagusnl** [comment]
Gotchu, thanks for the clarification!

File: CollapseLauncher/XAMLs/MainApp/Pages/GenshinGameSettingsPage.xaml.cs [hunk 1]
```diff
@@ -12,14 +12,26 @@
...
 using Hi3Helper;
 using Microsoft.UI.Xaml.Controls.Primitives;
 using Windows.Globalization.NumberFormatting;
+using Windows.Graphics.DirectX;
 using CollapseLauncher.Statics;
+using Microsoft.Graphics.Canvas.UI.Xaml;
+using Microsoft.Graphics.Canvas;
+using Microsoft.UI;
+using Windows.Storage.Streams;
+using Windows.Storage;
```
Comment (Cryotechnic) [inline CollapseLauncher/XAMLs/MainApp/Pages/GenshinGameSettingsPage.xaml.cs:31]: Really minor, but ideally, we'd like to group the imports together, such that imports are grouped by packages.

**Review by Cryotechnic — CHANGES_REQUESTED** [review]
Just a small change overall, the rest looks great. Thanks for your contribution!

**Comment by Cryotechnic** [comment]
Again, this is on SDR and is just a minor visual bug, but the second image does not fully render. I don't think it really matters since SDR display users will never enable this option, but just something to take note of.

Image in question:
image

The rest looks good!

File: CollapseLauncher/XAMLs/MainApp/Pages/GenshinGameSettingsPage.xaml [hunk 2]
```diff
@@ -81,75 +82,91 @@
...
+                                                </ToolTipService.ToolTip>
+                                            </xaml:CanvasSwapChainPanel>
                                             <Grid>
-                                                <TextBlock Text="{x:Bind helper:Locale.Lang._GenshinGameSettingsPage.Graphics_HDR_MaxLuminosity}" Style="{ThemeResource BodyStrongTextBlockStyle}" VerticalAlignment="Center"/>
-                                                <NumberBox Name="MaxLuminosityValue" HorizontalAlignment="Right" Minimum="300.0" Maximum="2000.0" NumberFormatter="{StaticResource DecimalFormatter}" CornerRadius="8,8,0,0" Margin="0,0,0,0" ValueChanged="MaxLuminosityValue_ValueChanged"/> 
+                                                <TextBlock Text="{x:Bind helper:Locale.Lang._GenshinGameSettingsPage.Graphics_HDR_MaxLuminosity}" Style="{ThemeResource BodyStrongTextBlockStyle}" VerticalAlignment="Center">
+                                                    <ToolTipService.ToolTip>
+                                                        <ToolTip Content="{x:Bind helper:Locale.Lang._GenshinGameSettingsPage.Graphics_HDR_MaxLuminosity_Help}"/>
```
Comment (bagusnl) [inline CollapseLauncher/XAMLs/MainApp/Pages/GenshinGameSettingsPage.xaml:145]: Same as above

File: CollapseLauncher/XAMLs/MainApp/Pages/GenshinGameSettingsPage.xaml [hunk 3]
```diff
@@ -81,75 +82,91 @@
...
                                             <Grid>
-                                                <TextBlock Text="{x:Bind helper:Locale.Lang._GenshinGameSettingsPage.Graphics_HDR_UiBrightness}" Style="{ThemeResource BodyStrongTextBlockStyle}" VerticalAlignment="Center"/>
-                                                <NumberBox Name="UiPaperWhiteValue" HorizontalAlignment="Right" VerticalAlignment="Center" Minimum="150.0" Maximum="550.0" NumberFormatter="{StaticResource DecimalFormatter}" CornerRadius="8,8,0,0" Margin="0,0,0,0" Height="32" ValueChanged="UiPaperWhiteValue_ValueChanged"/>
+                                                <TextBlock Text="{x:Bind helper:Locale.Lang._GenshinGameSettingsPage.Graphics_HDR_SceneBrightness}" Style="{ThemeResource BodyStrongTextBlockStyle}" VerticalAlignment="Center">
+                                                    <ToolTipService.ToolTip>
+                                                        <ToolTip Content="{x:Bind helper:Locale.Lang._GenshinGameSettingsPage.Graphics_HDR_SceneBrightness_Help}"/>
```
Comment (bagusnl) [inline CollapseLauncher/XAMLs/MainApp/Pages/GenshinGameSettingsPage.xaml:155]: Same as above

**Review by bagusnl — COMMENTED** [review]
Good work overall on the implementation!
Some things to discuss are:
1. Do we need to have tooltip if its already explained in the FlyOut?
2. Is there a way to tell user that they did not have HDR capable monitor?
The API to check if the HDR monitor is available can be done through [Windows.Graphics.Display.IsAdvancedColorKindAvailable](https://learn.microsoft.com/en-us/uwp/api/windows.graphics.display.advancedcolorinfo.isadvancedcolorkindavailable?view=winrt-22621)
This is to eliminate user confusion when the settings keeps being reverted and the image shown is blank

**Comment by shatyuka** [comment]
I have tried Windows.Graphics.Display.DisplayInformation, but it's only available in UWP app.
I did further research and it turns out I need to use **Microsoft**.Graphics.Display.DisplayInformation :(

**Comment by bagusnl** [comment]
I am so sorry I sent the wrong documentation :(

**Review by bagusnl — APPROVED** [review]
Good work on the overall implementation!
I have tested on Release build and everything seems good, the code looks good as well. 

Sorry for asking so much to you. Thank you so much for your contribution for Collapse project! <3

**Comment by Cryotechnic** [comment]
LGTM!

Thanks for your contributions! Going to merge now.

**Pull request merged**